### ISSUE TITLE
create: can't update page - mediaBox mismatch

### ISSUE THREAD

**Issue by LouisChering** [pdfcpu/pdfcpu#409]
I'm trying to use the new create package that was recently released but I'm getting either a "mediaBox mistmatch" or a "cropbox mistmatch".

The JSON payload I've used is the one from the example given by: (I've included this as example.txt)
``` sh
pdfcpu help create
```

When I run this test against the included "hello.pdf" or "multi-page-document.pdf" the error I described above.

``` golang

import (
	"fmt"
	"os"
	"testing"

	"github.com/pdfcpu/pdfcpu/pkg/api"
)

func Test_CreatePdfUsingJson(t *testing.T) {
	t.Parallel()

	jsonFile, err := os.Open("./example.txt")
	if err != nil {
		t.Log(err)
		t.FailNow()
	}
	defer jsonFile.Close()

	pdfFile, err := os.Open("./multi-page-document.pdf")
	if err != nil {
		t.Log(err)
		t.FailNow()
	}
	defer pdfFile.Close()

	outputFile, err := os.Create("./output.pdf")
	if err != nil {
		t.Log(err)
		t.FailNow()
	}
	defer outputFile.Close()

	err = api.CreateFromJSON(jsonFile, pdfFile, outputFile, nil)
	if err != nil {
		t.Log(err)
		t.FailNow()
	}
}
```
I've tried to step through the code and figure out whats going on.  It looks like if I comment out the lines below the test functions and I get some working output.

**pkg/pdfcpu/create/create.go (lines 109-115)** 
``` golang
	if !inhPAttrs.MediaBox.Equals(*p.MediaBox) {
		return errors.Errorf("pdfcpu: can't update page %d - mediaBox mismatch", currPageNr)
	}

	if !inhPAttrs.CropBox.Equals(*p.CropBox) {
		return errors.Errorf("pdfcpu: can't update page %d - cropBox mismatch", currPageNr)
	}
```

It looks like in some cases the CropBox check will fail because "inhPAttrs" has no CropBox assigned.  You can run into this error by tryting to process hello.pdf and commenting out the MediaBox.Equals check.

[hello.pdf](https://github.com/pdfcpu/pdfcpu/files/7705552/hello.pdf) - This is a PDF I've typed up myself
[multi-page-document.pdf](https://github.com/pdfcpu/pdfcpu/files/7705556/multi-page-document.pdf) - This is a PDF generated by Google Drive
[example.txt](https://github.com/pdfcpu/pdfcpu/files/7705563/example.txt)

As a side note, the text that is appended to the multi-page-document.pdf appears to be upside down as well.

**Comment by hhrutter** [comment]
Quick feedback: 
* Right now *create* for updating an existing PDF works best with files produced by pdfcpu
* Per definition the mediaBox of the input file has to match the mediaBox you specify in JSON
* The cropBox check might be buggy

**Comment by LouisChering** [comment]
**"Right now create for updating an existing PDF works best with files produced by pdfcpu"**
Thank you for clarifying this.  Does this statement cover documents that have been optimised using pdfcpu?

i.e If I start with a document that was created outside of pdfcpu and run it through decryption with optimisation enabled.  Would you expect those documents to eventually work with the _create_ API.

**Comment by hhrutter** [comment]
Sorry I can't really tell - *Create* is really an Alpha feature since part of it is form generation at that is coming up for the next release. By then *Create* will be more flexible and robust.

For now I would stick to creating files from scratch and then repeatedly add content and wait for the next release for appending content to existing files via *Create*.

**Comment by hhrutter** [comment]
This should be fixed with v0.4.0
